docusaurus/docs/React/guides/date-time-formatting.mdx

```diff
@@ -0,0 +1,208 @@
+>
+  ...
+</Channel>;
+```
+
+### Custom formatting function
+
+Custom formatting function can be passed to `MessageList` or `VirtualizedMessageList` via prop `formatDate` (`(date: Date) => string;`). The `Message` component passes down the function to be consumed by the children via `MessageComponentContext`:
+
+```jsx
+import { useMessageContext } from 'stream-chat-react';
+const CustomComponent = () => {
+  const { formatDate } = useMessageContext();
+};
+```
+
+By default, the function is consumed by the `MessageTimestamp` component. This means the formatting via `formatDate` is reduced only to timestamp shown by a message in the message list. Components `DateSeparator`, `EventComponent` would ignore the custom formatting.
+
+### Date & time formatting with i18n service
+
+Until now, the datetime values could be customized within the `Channel` component at best. Formatting via i18n service allows for SDK wide configuration. The configuration is stored with other translations in JSON files. Formatting with i18n service has the following advantages:
+
+- it is centralized
+- it takes into consideration the locale out of the box
+- allows for high granularity - formatting per string, not component (opposed to props approach)
+- allows for high re-usability - apply the same configuration in multiple places via the same translation key
+- allows for custom formatting logic
+
+#### Change the default configuration
+
+The default datetime formatting configuration is stored in the JSON translation files. The default translation keys are namespaced with prefix `timestamp/` followed by the component name. For example, the message date formatting can be targeted via `timestamp/MessageTimestamp`, because the underlying component is called `MessageTimestamp`.
+
+##### Overriding the prop defaults
+
+The default date and time rendering components in the SDK were created with default prop values that override the configuration parameters provided over JSON translations. Therefore, if we wanted to configure the formatting from JSON translation files, we need to nullify the prop defaults first. An example follows:
+
+```tsx
+import {
+  DateSeparatorProps,
+  DateSeparator,
+  EventComponentProps,
+  EventComponent,
+  MessageTimestampProps,
+  MessageTimestamp,
+} from 'stream-chat-react';
+
+const CustomDateSeparator = (props: DateSeparatorProps) => (
+  <DateSeparator {...props} calendar={null} /> // calendarFormats, neither format have default value
+);
+
+const SystemMessage = (props: EventComponentProps) => (
+  <EventComponent {...props} format={null} /> // calendar  neither calendarFormats have default value
+);
+
+const CustomMessageTimestamp = (props: MessageTimestampProps) => (
+  <MessageTimestamp {...props} calendar={null} format={null} /> // calendarFormats do not have default value
+);
+```
+
+Now we can apply custom configuration in all the translation JSON files. It could look similar to the following example key-value pair.
```

Comment:
```suggestion
Now we can apply custom configuration in all the translation JSON files. It could look similar to the following key-value pair example.
```